tajo-catalog/tajo-catalog-client/src/main/java/org/apache/tajo/catalog/AbstractCatalogClient.java

```
@@ -140,13 +143,14 @@ public final Boolean existTablespace(final String tablespaceName) {
   }
 
   @Override
-  public TablespaceProto getTablespace(final String tablespaceName) {
+  public TablespaceProto getTablespace(final String tablespaceName) throws UndefinedTableException {
 
     try {
       final BlockingInterface stub = getStub();
       final GetTablespaceResponse response = stub.getTablespace(null, ProtoUtil.convertString(tablespaceName));
-      ensureOk(response.getState());
 
+      throwsIfThisError(response.getState(), UndefinedTableException.class);
```

This method should throw UndefinedTablespaceException.